src/fu-engine.c

```diff
@@ -756,6 +756,8 @@ fu_engine_set_release_from_appstream(FuEngine *self,
 			fwupd_release_set_update_image(rel, tmp);
 		}
 	}
+	if (xb_node_get_attr(release, "date_eol") != NULL)
```

Comment:
isn't it `is_eol`?

Reply:
Nah, it's a ISO date in https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases

Reply:
Oh got it.  In that case is it worth changing this all around a bit and letting OEMs set a "future" EOL date that can be reflected in the GUI/tool?

I think the supported flag can become either:
* Supported (when no EOL date is set)
* Supported until $DATE (when one is set in future)
* End of Life since $DATE (when one was set in past)

Reply:
> and letting OEMs set a "future" EOL date

I think I'm going to struggle getting them to click a big red button, to be honest. I put the LVFS mtime as a DateTime deliberately (rather than a bool is_eol) so we can support this in the future.

Reply:
Maybe we also want a `fwupd_device_get_end_of_life_date()` or something like that? Not sure about the name tho :/